packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/UnbrandedTypeSpecClient.cs

```diff
@@ -15,89 +17,435 @@ internal void CreateSayHiRequest(Uri unbrandedTypeSpecUrl, string headParameter,
             throw new NotImplementedException("Method not implemented.");
         }
 
+        /// <summary> Return hi. </summary>
+        /// <param name="headParameter"></param>
+        /// <param name="queryParameter"></param>
+        /// <param name="optionalQuery"></param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="headParameter"/> or <paramref name="queryParameter"/> is null. </exception>
+        public virtual void SayHi(string headParameter, string queryParameter, string optionalQuery, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(headParameter, nameof(headParameter));
+            Argument.AssertNotNull(queryParameter, nameof(queryParameter));
+
+        }
+
+        /// <summary> Return hi. </summary>
+        /// <param name="headParameter"></param>
+        /// <param name="queryParameter"></param>
+        /// <param name="optionalQuery"></param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="headParameter"/> or <paramref name="queryParameter"/> is null. </exception>
+        public virtual async void SayHiAsync(string headParameter, string queryParameter, string optionalQuery, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(headParameter, nameof(headParameter));
+            Argument.AssertNotNull(queryParameter, nameof(queryParameter));
+
+            await Task.CompletedTask.ConfigureAwait(false);
+        }
+
         internal void CreateHelloAgainRequest(Uri unbrandedTypeSpecUrl, string p1, string contentType, string p2, RoundTripModel action, string accept)
         {
             throw new NotImplementedException("Method not implemented.");
         }
 
+        /// <summary> Return hi again. </summary>
+        /// <param name="p1"></param>
+        /// <param name="p2"></param>
+        /// <param name="action"></param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="p1"/>, <paramref name="p2"/> or <paramref name="action"/> is null. </exception>
+        public virtual void HelloAgain(string p1, string p2, RoundTripModel action, CancellationToken cancellationToken = default)
```

Comment:
It looks like this is a convenience method. I see that we don't take a CancellationToken parameter in the typespec-generated code.

Is this PR just to refactor MethodProviderCollection, and will you have a separate PR to generate protocol methods?

Reply:
this is just a wip and will be fixed before the issue closes but allows others to work off of this refactor